lib/bolt/cli.rb

```diff
@@ -525,14 +523,23 @@ def in_bolt_compiler(opts = [])
           begin
             yield compiler
           rescue Puppet::PreformattedError => err
-            err.cause
-          rescue StandardError => err
+            # In what cases does this help?
+            if err.cause && err.cause.message
+              Bolt::CLIError.new(err.cause.message)
```

Comment:
@MikaelSmith in what cases do we want to use the causes message? Its causing at least some errors to be squashed.

Reply:
I tested some basic Puppet parsing errors, and errors thrown by Bolt functions. Specific errors I tried to unwrap were: `raise 'failure'`, `raise <Bolt::Error>`, and `raise <Ruby Error>`.